src/test/java/fr/sncf/osrd/train/StopTests.java

```diff
@@ -66,6 +67,47 @@ public void testStopDuration() throws InvalidInfraException {
         assertEquals(timeShort - durationStopShort, timeLong - durationStopLong, 0.1);
     }
 
+    public void testStopDurationNull() throws InvalidInfraException {
+        final var infra = getBaseInfra();
+        final var configStop = makeConfigWithGivenStops("tiny_infra/config_railjson.json",
+                new RJSTrainStop[]{
+                        new RJSTrainStop(2000., null, 100),
+                        new RJSTrainStop(1000., null, 0),
+                        new RJSTrainStop(-1., null, 0)
+                }
+        );
+        final var configNoStop = makeConfigWithGivenStops("tiny_infra/config_railjson.json",
+                new RJSTrainStop[]{
+                        new RJSTrainStop(2000., null, 100),
+                        new RJSTrainStop(-1., null, 0)
+                }
+        );
+        var simStop = Simulation.createFromInfraAndEmptySuccessions(RailJSONParser.parse(infra), 0, null);
+        run(simStop, configStop);
+        var timeWithStop = simStop.getTime();
+
+        var simNoStop = Simulation.createFromInfraAndEmptySuccessions(RailJSONParser.parse(infra), 0, null);
+        run(simNoStop, configNoStop);
+        var timeNoStop = simNoStop.getTime();
+
+        assertEquals(timeNoStop, timeWithStop, 0.1);
+    }
+
+    public void testStopEndDurationNull() throws InvalidInfraException {
+        final var infra = getBaseInfra();
+        final var config = makeConfigWithGivenStops("tiny_infra/config_railjson.json",
+                new RJSTrainStop[]{
+                        new RJSTrainStop(1000., null, 0),
+                        new RJSTrainStop(-1., null, 0)
+                }
+        );
+
+        var sim = Simulation.createFromInfraAndEmptySuccessions(RailJSONParser.parse(infra), 0, null);
+        run(sim, config);
+
+        assertTrue(sim.isSimulationOver());
```

Comment:
This doesn't really test anything, the simulation is always over after `run()`. You could test that the train last speed and fail if it's below a certain value.